### ISSUE TITLE
fails to build with clang

### ISSUE THREAD

**Issue by zetok** [irungentoo/toxcore#1508]
https://travis-ci.org/tux3/qTox/jobs/104836206
↑ fail, note that other things (e.g. ffmpeg, filter_audio) compile just fine

Same env, builds with GCC 4.8: https://travis-ci.org/tux3/qTox/jobs/104836205

**Comment by GrayHatter** [comment]
How is this a core issue?

 It's not failing to build, just install. 

It only occurs on qTox

And you already have a working Travis build?

**Comment by zetok** [comment]
>How is this a core issue?

How it's not a core issue? I.e. if building & installing toxcore aren't toxcore problems, then where it should be reported?

>It's not failing to build, just install. 

Right. Thanks for pointing that out.

>It only occurs on qTox

Could you please elaborate on that? Build script didn't even get to qTox compiling part, since it failed to build & install ***toxcore***.

>And you already have a working Travis build?

Yes, I have a working build with *GCC*. I've linked to it to show that in the same environment, spare for the compiler, things work.

You know, if clang isn't supported by toxcore, just slap a label with that info somewhere (e.g. in INSTALL.md would be nice), so that no one will waste time even trying to use it to build toxcore.

**Comment by GrayHatter** [comment]
https://travis-ci.org/irungentoo/toxcore/builds/96489160

Clang works here...

**Comment by zetok** [comment]
>https://travis-ci.org/irungentoo/toxcore/builds/96489160

>Clang works here...

Does it? Where do you see `make install` for toxcore? Not to mention that it's a build from 2 months ago..

**Comment by nurupo** [comment]
Yeah, I saw Travis to fail at `make install` of toxcore when I was playing around while updating the Travis script for toxcore a few days ago. Not sure how to fix it, maybe @jin-eld or @irungentoo would know.

**Comment by jin-eld** [comment]
Hmm, I tried it here (Fedora 23) with clang version 3.7.0 (tags/RELEASE_370/final), I used:

CC=clang ./configure --prefix=/mydir && make -j4 && make install 

It did work without errors, everything got installed... I later also tried your configure line with the extra CFLAGS and so on, but still had no problem running make && make install
